core/textile/bucket_factory.go

```diff
@@ -200,11 +201,23 @@ func (tc *textileClient) createBucket(ctx context.Context, bucketSlug string) (B
 
 	// We store the bucket in a meta thread so that we can later fetch a list of all buckets
 	log.Debug("Bucket " + bucketSlug + " created. Storing metadata.")
-	_, err = tc.getModel().CreateBucket(ctx, bucketSlug, dbID.String())
+	schema, err := m.CreateBucket(ctx, bucketSlug, dbID.String())
+	if err != nil {
+		return nil, err
+	}
+
+	mirrorSchema, err := tc.createMirrorBucket(ctx, *schema)
 	if err != nil {
 		return nil, err
 	}
 
+	if mirrorSchema != nil {
+		_, err = m.CreateMirrorBucket(ctx, bucketSlug, mirrorSchema)
```

Comment:
It is confusing that this function is calling both `model.CreateMirrorBucket` and `tc.CreateMirrorBucket`. Maybe the collection creation can be taken care of inside tc.createMirrorBucket`?